internal/inputs/generator_queries.go

```diff
@@ -0,0 +1,272 @@
+	"github.com/timescale/tsbs/cmd/tsbs_generate_queries/databases/siridb"
+	"github.com/timescale/tsbs/cmd/tsbs_generate_queries/databases/timescaledb"
+	"github.com/timescale/tsbs/cmd/tsbs_generate_queries/utils"
+)
+
+// Error messages when using a QueryGenerator
+const (
+	ErrInvalidQueryConfig = "invalid config: QueryGenerator needs a QueryGeneratorConfig"
+	ErrEmptyQueryType     = "query type cannot be empty"
+
+	errBadQueryTypeFmt        = "invalid query type for use case '%s': '%s'"
+	errCouldNotDebugFmt       = "could not write debug output: %v"
+	errCouldNotEncodeQueryFmt = "could not encode query: %v"
+	errCouldNotQueryStatsFmt  = "could not output query stats: %v"
+)
+
+// QueryGeneratorConfig is the GeneratorConfig that should be used with a
+// QueryGenerator. It includes all the fields from a BaseConfig, as well as
+// options that are specific to generating the queries to test against a
+// database, such as the query type and individual database options.
+type QueryGeneratorConfig struct {
+	BaseConfig
+	QueryType            string
+	InterleavedGroupID   uint
+	InterleavedNumGroups uint
+
+	// TODO - I think this needs some rethinking, but a simple, elegant solution escapes me right now
+	TimescaleUseJSON       bool
+	TimescaleUseTags       bool
+	TimescaleUseTimeBucket bool
+
+	ClickhouseUseTags bool
+
+	MongoUseNaive bool
+}
+
+// Validate checks that the values of the QueryGeneratorConfig are reasonable.
+func (c *QueryGeneratorConfig) Validate() error {
+	err := c.BaseConfig.Validate()
+	if err != nil {
+		return err
+	}
+
+	if c.QueryType == "" {
+		return fmt.Errorf(ErrEmptyQueryType)
+	}
+
+	err = validateGroups(c.InterleavedGroupID, c.InterleavedNumGroups)
+	return err
+}
+
+func (c *QueryGeneratorConfig) AddToFlagSet(fs *flag.FlagSet) {
+	c.BaseConfig.AddToFlagSet(fs)
+	flag.StringVar(&c.QueryType, "query-type", "", "Query type. (Choices are in the use case matrix.)")
+
+	flag.UintVar(&c.InterleavedGroupID, "interleaved-generation-group-id", 0,
+		"Group (0-indexed) to perform round-robin serialization within. Use this to scale up data generation to multiple processes.")
+	flag.UintVar(&c.InterleavedNumGroups, "interleaved-generation-groups", 1,
+		"The number of round-robin serialization groups. Use this to scale up data generation to multiple processes.")
+
+}
+
+// QueryGenerator is a type of Generator for creating queries to test against a
+// database. The output is specific to the type of database (due to each using
+// different querying techniques, e.g. SQL or REST), but is consumed by TSBS
+// query runners like tsbs_run_queries_timescaledb.
+type QueryGenerator struct {
+	// Out is the writer where data should be written. If nil, it will be
+	// os.Stdout unless File is specified in the GeneratorConfig passed to
+	// Generate.
+	Out io.Writer
+	// Out2 is where non-generated data should be written. If nil, it will be
+	// os.Stderr.
+	Out2 io.Writer
```

Comment:
I feel like this might be a little annoying to read for newcomers. Maybe something like `DataOutput` and `LogOutput` or `InfoOutput` ?